text/0000-convert-id.md

```diff
@@ -0,0 +1,203 @@
+- Feature Name: convert_identity
+- Start Date: 2018-01-19
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Adds an identity function `pub const fn identity<T>(x: T) -> T { x }` as
+`core::convert::identity`. The function is also re-exported to
+`std::convert::identity` as well as the prelude of
+both libcore and libstd.
+
+# Motivation
+[motivation]: #motivation
+
+## The identity function is useful
+
+While it might seem strange to have a function that just returns back the input,
+there are some cases where the function is useful.
+
+### Using `identity` to do nothing among a collection of mappers
+
+When you have collections such as maps or arrays of mapping functions like
+below and you watch to dispatch to those you sometimes need the identity
+function as a way of not transforming the input. You can use the identity
+function to achieve this.
+
+```rust
+// Let's assume that this and other functions do something non-trivial.
+fn do_interesting_stuff(x: u32) -> u32 { .. }
+
+// A dispatch-map of mapping functions:
+let mut map = HashMap::new();
+map.insert("foo", do_interesting_stuff);
+map.insert("bar", other_stuff);
+map.insert("baz", identity);
+```
+
+### Using `identity` as a no-op function in a conditional
+
+This reasoning also applies to simpler yes/no dispatch as below:
+
+```rust
+let mapper = if condition { some_manipulation } else { identity };
+
+// do more interesting stuff inbetween..
+
+do_stuff(42);
+```
+
+### Using `identity` to concatenate an iterator of iterators
+
+We can use the identity function to concatenate an iterator of iterators
+into a single iterator.
```

Comment:
This section argues to me more that we should have a `flatten` or [`concat`](http://sml-family.org/Basis/list.html#SIG:LIST.concat:VAL) or something, since I don't find `flat_map(identity)` any clearer than `flat_map(|x| x)`—neither conveys the intent, to me.

Reply:
And now `.flatten()` is in your nightly compiler in a day or so. Tracking issue: https://github.com/rust-lang/rust/issues/48213